lib/tasks/fix_non_en_redirects.rake

```diff
@@ -0,0 +1,85 @@
+namespace :fix_non_en_redirects do
+  desc "Fix non-en redirects for content, filtering by content_id."
+  namespace :by_content_id do
+    def fix_draft_redirect(draft_redirect_edition, correct_locale, dry_run:)
+      if dry_run
+        puts "Would change Document #{draft_redirect_edition.document.id} from locale: #{draft_redirect_edition.document.locale} to locale: #{correct_locale}"
+
+      else
+        # This is really bad, as it's not going through the API of
+        # the Publishing API, but unfortunately I don't see a
+        # neater way of fixing this issue with the Document having
+        # the wrong locale.
+        draft_redirect_edition
+          .document
+          .update!(
+            locale: correct_locale,
+          )
+
+        publish_payload = {
+          content_id: draft_redirect_edition.document.content_id,
+          locale: correct_locale,
+          update_type: "major", # Some redirects may be missing an
+          # update_type, so work around that
+          # by specifying one here
+        }
+
+        Commands::V2::Publish.call(publish_payload)
+      end
+    end
+
+    def fix_for_content_id(content_id, dry_run:)
+      document_ids = Document.where(content_id: content_id).pluck(:id)
+
+      relevant_base_paths = Edition.where(
+        document_id: document_ids,
+      ).where.not(
+        document_type: "redirect",
+      ).select(:base_path)
+
+      draft_redirect_editions = Edition.where(
+        document_type: "redirect",
+        state: "draft",
+        base_path: relevant_base_paths,
+      ).order(:id)
+
+      draft_redirect_editions.each do |draft_redirect_edition|
+        editions_matching_destination = Edition.where(
+          base_path: Edition.find(3478439).redirects.first[:destination],
```

Comment:
What does this edition refer to?

Reply:
Haha, it's the edition I was testing with. In the general case it should be the draft redirect edition though, as it's the destination of the draft redirect that you want to use in the query.

I've fixed this now.